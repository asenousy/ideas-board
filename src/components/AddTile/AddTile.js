import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "./AddTile.module.css";
import { faPlusCircle } from "@fortawesome/free-solid-svg-icons";

export const AddTile = props => {
  return (
    <button className={styles.wrapper} onClick={props.onClick} aria-label="new">
      <FontAwesomeIcon className={styles.add} icon={faPlusCircle} size="3x" />
    </button>
  );
};

AddTile.propTypes = {
  onClick: PropTypes.func
};

export default React.memo(AddTile);
