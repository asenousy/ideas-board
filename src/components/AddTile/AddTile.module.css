.wrapper {
  box-sizing: border-box;
  height: 100px;
  width: 100px;
  margin: 10px;
  border-radius: 5px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  background-color: white;
}

.add {
  color: green;
}
