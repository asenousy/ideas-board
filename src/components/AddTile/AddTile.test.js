import React from "react";
import { shallow } from "enzyme";
import AddTile from "./AddTile";

let wrapper;
beforeEach(() => (wrapper = shallow(<AddTile />)));

it("renders without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

it("calls onClick when click event", () => {
  const mockFn = jest.fn();
  wrapper.setProps({ onClick: mockFn });
  wrapper.simulate("click");
  expect(mockFn.mock.calls.length).toBe(1);
});
