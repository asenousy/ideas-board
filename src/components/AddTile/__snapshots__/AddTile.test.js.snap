// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "new",
      "children": <FontAwesomeIcon
        border={false}
        className="add"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f055",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
            ],
            "iconName": "plus-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />,
      "className": "wrapper",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "border": false,
        "className": "add",
        "fixedWidth": false,
        "flip": null,
        "icon": Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f055",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
          ],
          "iconName": "plus-circle",
          "prefix": "fas",
        },
        "inverse": false,
        "listItem": false,
        "mask": null,
        "pull": null,
        "pulse": false,
        "rotation": null,
        "size": "3x",
        "spin": false,
        "symbol": false,
        "title": "",
        "transform": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "new",
        "children": <FontAwesomeIcon
          border={false}
          className="add"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f055",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
              ],
              "iconName": "plus-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />,
        "className": "wrapper",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "add",
          "fixedWidth": false,
          "flip": null,
          "icon": Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f055",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
            ],
            "iconName": "plus-circle",
            "prefix": "fas",
          },
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": "3x",
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
